File: packages/dd-trace/test/llmobs/parity/scenarios/anthropic/beta-messages.py
import os

from anthropic import Anthropic

from common import finish, start


start("anthropic")
client = Anthropic(api_key=os.environ["ANTHROPIC_API_KEY"], base_url=os.environ["PROVIDER_BASE_URL"])
client.beta.messages.create(
    model="claude-3-5-sonnet-20241022",
    max_tokens=16,
    messages=[{"role": "user", "content": "Hello from the beta parity harness."}],
)
finish()

File: packages/dd-trace/test/llmobs/parity/scenarios/anthropic/beta-messages.js
'use strict'

const { finish, loadTracer, loadVersionedModule } = require('../common')

async function main () {
  const tracer = loadTracer('anthropic')
  const { Anthropic } = loadVersionedModule('@anthropic-ai/sdk')
  const client = new Anthropic({ apiKey: process.env.ANTHROPIC_API_KEY, baseURL: process.env.PROVIDER_BASE_URL })
  await client.beta.messages.create({
    model: 'claude-3-5-sonnet-20241022',
    max_tokens: 16,
    messages: [{ role: 'user', content: 'Hello from the beta parity harness.' }],
  })
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack ?? error}\n`)
  process.exitCode = 1
})
